File: frontend/src/app/models/Task.ts
import { Employee } from './Employee';
import { Status } from './Status';
import { Project } from './Project';

export class Task {
  idTask: number =0;
  title: string ='';
  description: string='';
  dueDate: Date = new Date();
  startDate: Date = new Date();
  endDate: Date= new Date();
  status: Status = Status.TO_DO;
  project: Project = new Project();
  assignedEmp: Employee = new Employee();
}

File: backend/src/main/java/tn/company/taskManagement/entities/Task.java
package tn.company.taskManagement.entities;

import jakarta.persistence.*;
import java.util.Date;
import lombok.*;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Task {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private int idTask;

        private String title;
        private String description;

        @Temporal(TemporalType.TIMESTAMP)
        private Date dueDate;

        @Temporal(TemporalType.TIMESTAMP)
        private Date startDate;

        @Temporal(TemporalType.TIMESTAMP)
        private Date endDate;

        @Enumerated(EnumType.STRING)
        private Status status;

        @ManyToOne
        @JoinColumn(name = "project_id")
        private Project project;

        @ManyToOne
        @JoinColumn(name = "user_id")
        private Employee assignedEmp;
}
